### PR TITLE
[SparkR][Minor] Add more examples to window function docs

### PR DESCRIPTION
**Pull request by junyangq**
## What changes were proposed in this pull request?

This PR adds more examples to window function docs to make them more accessible to the users.

It also fixes default value issues for `lag` and `lead`.
## How was this patch tested?

Manual test, R unit test.

### CODE REVIEW

File: R/pkg/R/functions.R [hunk 1]
```diff
@@ -3239,7 +3259,15 @@ setMethod("lead",
...
 #' @family window_funcs
 #' @export
-#' @examples \dontrun{ntile(1)}
+#' @examples \dontrun{
+#'   df <- createDataFrame(iris)
+#'
+#'   # Partition by Species and order by Sepal_Length
+#'   ws <- orderBy(windowPartitionBy("Species"), "Sepal_Length")
+#'
+#'   # Get ntil group id (1-4) for Sepal_Length
```
Comment (felixcheung) [inline R/pkg/R/functions.R]: ntile?

**Comment by felixcheung** [comment]
btw, you might want to avoid using `iris` for example. There has been several bugs where people would like to address the limitation with column names with `.` - if that is fixed then the example would be wrong to reference column as `Sepal_Length` instead of its original name `Sepal.Length`

File: R/pkg/R/functions.R [hunk 2]
```diff
@@ -3200,19 +3212,27 @@ setMethod("lag",
 #' This is equivalent to the \code{LEAD} function in SQL.
 #'
 #' @param x Column to compute on
-#' @param offset Number of rows to offset
+#' @param offset Number of rows to offset. If not specified, the default is 1.
```
Comment (felixcheung) [inline R/pkg/R/functions.R]: perhaps explain more what offset/defaultValue will do like in Python?

Comment (junyangq) [inline R/pkg/R/functions.R]: This is actually in the description?

File: R/pkg/R/functions.R [hunk 3]
```diff
@@ -3194,25 +3206,34 @@ setMethod("lag",
...
 #' @aliases lead,characterOrColumn,numeric-method
 #' @export
-#' @examples \dontrun{lead(df$c)}
+#' @examples \dontrun{
+#'   df <- createDataFrame(mtcars)
+#'
+#'   # Partition by am (transmission) and order by hp (horsepower)
+#'   ws <- orderBy(windowPartitionBy("am"), "hp")
+#'
+#'   # Lag mpg values by 1 row on the partition-and-ordered table
```
Comment (felixcheung) [inline R/pkg/R/functions.R]: lag -> lead?

File: R/pkg/R/functions.R (hunk 2 above)
Comment (felixcheung) [inline R/pkg/R/functions.R]: in @rdname lag, there's a bit more information on these @param. Perhaps here with @rdname lead we could do similar?

**Comment by felixcheung** [comment]
LGTM. waiting for Jenkins.

**Pull request closed without merging**